### PR TITLE
Add gapless assembly improvement tool

### PR DESCRIPTION
**Pull request by schmeing**
Adding package for gapless. A python only assembly improvement tool that combines scaffolding, gap-closing and assembly correction with long reads.

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

For members of the Bioconda project, the following command is also available:

<table>
  <tr>
    <td><code>@BiocondaBot please merge</code></td>
    <td>Upload built packages/containers and merge a PR. <br />Someone must approve a PR first! <br />This reduces CI build time by reusing built artifacts.</td>
  </tr>
</table>

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

**Comment by schmeing** [comment]
@BiocondaBot please add label

File: recipes/gapless/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    - numpy
+    - scipy
+    - seaborn
+    - matplotlib-base
+    - pillow
+    - biopython
+
+test:
+  commands:
+    - gapless.py
```
Comment (kdm9) [inline recipes/gapless/meta.yaml:26]: maybe run gapless.sh as well as gapless.py to check both are available

File: recipes/gapless/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    - gapless.py 2>&1 | grep "https://github.com/schmeing/gapless"
+
+about:
+  home: https://github.com/schmeing/gapless
+  license: MIT
+  summary: gapless assembly improvemen tool
```
Comment (kdm9) [inline recipes/gapless/meta.yaml:32]: missing t from improvemen**t**.

**Review by kdm9 — CHANGES_REQUESTED** [review]
Only two minor changes. (NB: i'm not a bioconda maintainer, but I was about to make a conda package for gapless before realising there was already a draft)

**Comment by kdm9** [comment]
@BiocondaBot please update

**Comment by kdm9** [comment]
@BiocondaBot please fetch artifacts

**Comment by priesgo** [comment]
Would you, please, add URLs to the original repository in the description?

**Comment by priesgo** [comment]
I saw that the specific versions of the python dependencies are described in the gapless readme file. It would be good to have those specified in the bioconda recipe.

**Comment by priesgo** [comment]
I was amazed that gapless python code is in a single file of 1.42 MB. How many lines of code does that have? Are there any tests implemented?
I don't know if quality of code is a legitimate requirement for inclusion in bioconda though. I would love to hear others opinions.

**Comment by priesgo** [comment]
Also the python interpreter tested is reported in the readme 3.9, but in the recipe 3.6.

**Comment by kdm9** [comment]
> I was amazed that gapless python code is in a single file of 1.42 MB. How many lines of code does that have? Are there any tests implemented? I don't know if quality of code is a legitimate requirement for inclusion in bioconda though. I would love to hear others opinions.

Yes, we would all like a system in which there was any reward of well-engineered academic software. Back in reality, gapless works well enough despite its unusual state. It does seem to contain tests (I'm not the author, just a user who needs a conda pkg).

**Comment by priesgo** [comment]
I understand. I'm fine with it if no one opposes, it was mostly surprise.
Nevertheless, the comments on the dependency versions are related to the recipe.

**Comment by schmeing** [comment]
In the readme I put my newest test environment at the time I wrote the readme. That does not state that versions before do not work. For python 3.6 I do have a test environment and it definitely works. I added the minimum version for pandas, where I am aware that versions before it do not work. For all other dependencies I am not aware of issues with specific versions. I can of course limit it strictly to versions I tested, but that would make it very hard to install it together with other packages. Thus, I would suggest to keep it open until incompatibilities have been reported.

**Review by kdm9 — APPROVED** [review]
Looks good to me, thanks @schmeing

File: recipes/gapless/meta.yaml (hunk 1 above)
Comment (kdm9) [reply recipes/gapless/meta.yaml:26]: (resolved below)

**Review by priesgo — APPROVED** [review]

**Comment by priesgo** [comment]
Thanks @schmeing, I agree let's try to get out there a first working version and continue from there. I have seen before that when no upper boundary is set the package stops installing properly suddenly due to a new release of a dependency that is not compatible anymore. Let's see how this works for you.

**Comment by priesgo** [comment]
@BiocondaBot please merge

**Pull request merged**